aten/src/ATen/functorch/BatchRulesScatterOps.cpp

```diff
@@ -1056,6 +1056,165 @@ std::tuple<Tensor,optional<int64_t>> masked_fill_scalar_batch_rule(
   return std::make_tuple(result, 0);
 }
 
+std::tuple<Tensor,optional<int64_t>> index_fill_int_scalar_batch_rule_impl(
+    Tensor & self, optional<int64_t> self_bdim,
+    int64_t dim,
+    const Tensor & index, optional<int64_t> index_bdim,
+    const Scalar & value,
+    const bool inplace) {
+    const auto self_logical_rank = rankWithoutBatchDim(self, self_bdim);
+    const auto index_logical_rank = rankWithoutBatchDim(index, index_bdim);
+    Tensor self_ = moveBatchDimToFront(self, self_bdim);
+    Tensor index_ = moveBatchDimToFront(index, index_bdim);
+    dim = maybe_wrap_dim(dim, self_logical_rank);
+
+    if (inplace && !self_bdim.has_value()) {
+      vmapIncompatibleInplaceError("index_fill_");
+    }
+
```

Comment:
Some style guide nits: the intents should be at different levels, otherwise it gets a bit difficult to read:

```
std::tuple<Tensor,optional<int64_t>> index_fill_int_scalar_batch_rule_impl(
    Tensor & self, optional<int64_t> self_bdim,
    Tensor & self, optional<int64_t> self_bdim,
    int64_t dim,
    const Tensor & index, optional<int64_t> index_bdim,
    const Scalar & value,
    const bool inplace) {
  const auto self_logical_rank = rankWithoutBatchDim(self, self_bdim);
  const auto index_logical_rank = rankWithoutBatchDim(index, index_bdim);
  Tensor self_ = moveBatchDimToFront(self, self_bdim);
  ...
}
```